lunabot_control/scripts/differential_drive_controller.py

```diff
@@ -13,20 +13,16 @@ def __init__(self):
         # ROS Publishers and subsribers to get / send data
 
         self._vel_sub = rospy.Subscriber("/cmd_vel", Twist, self._vel_cb)
-        self._right_drive_pub = rospy.Publisher("/right_drive", Int8, queue_size=1)
-        self._left_drive_pub = rospy.Publisher("/left_drive", Int8, queue_size=1)
+        self._right_drive_pub = rospy.Publisher("/right_drive", Int32, queue_size=1)
+        self._left_drive_pub = rospy.Publisher("/left_drive", Int32, queue_size=1)
         self._state_sub = rospy.Subscriber("sensors", RobotSensors, self._robot_state_cb)
 
         self.width = rospy.get_param("~width", 0.5588)
         self.max_speed_percentage = rospy.get_param("~max_speed_percentage", 0.8)
         self.hz = rospy.get_param("~hz", 20)
 
-        self._max_speed = rospy.get_param("~max_speed", 2.0) # In rad/s
-        self.p = rospy.get_param("~p", 3.9)  # P gain for PID controller
-        self.i = rospy.get_param("~i", 0.0)  # I gain for PID controller
-        self.d = rospy.get_param("~d", 0)  # D gain for PID controller
-        self.i_sat = rospy.get_param("~i_saturate", 10)  # Max for integral term
-
+        self._max_speed = rospy.get_param("~max_speed", 2.0) * 60 / 2 / np.pi # In rad/s converted to RPM
```

Comment:
Is there a reason why this isn't in RPM directly? Also, to convert from linear to angular speed, don't you need wheel diameter (12 inches, if it's relevant)

Reply:
Max RPM speed is 3000 at full torque and 5000 overall. I see very little downside to capping this at 3000 rpm